crypto/ec/ec_lcl.h

```diff
@@ -637,4 +646,73 @@ int X25519(uint8_t out_shared_key[32], const uint8_t private_key[32],
 void X25519_public_from_private(uint8_t out_public_value[32],
                                 const uint8_t private_key[32]);
 
+/*-
+ * This functions computes a single point multiplication over the EC group,
+ * using, at a high level, a Montgomery ladder with conditional swaps, with
+ * various timing attack defenses.
+ *
+ * It performs either a fixed point multiplication
+ *          (scalar * generator)
+ * when point is NULL, or a variable point multiplication
+ *          (scalar * point)
+ * when point is not NULL.
+ *
+ * scalar cannot be NULL and should be in the range [0,n) otherwise all constant
+ * time bets are off (where n is the cardinality of the EC group).
+ *
+ * NB: This says nothing about the constant-timeness of the ladder step
+ * implementation (i.e., the default implementation is based on EC_POINT_add and
+ * EC_POINT_dbl, which of course are not constant time themselves) or the
+ * underlying multiprecision arithmetic.
+ *
+ * The product is stored in r.
+ *
+ * This is an internal function: callers are in charge of ensuring that the
+ * input parameters `group`, `r`, `scalar` and `ctx` are not NULL.
+ *
+ * Returns 1 on success, 0 otherwise.
+ */
+int ec_mul_consttime(const EC_GROUP *group, EC_POINT *r,
+                     const BIGNUM *scalar, const EC_POINT *point,
+                     BN_CTX *ctx);
```

Comment:
I for one sense inconsistency in this function naming. Now when it's shared, a better name might be more appropriate. Maybe ec_mul_by_scalar_consttime? Just me? Obviously side note...

Reply:
What about `ec_scalarmul_ladder`?

I feel a bit of resistance in naming anything "consttime" given that we cannot promise anything about the underlying layer.

Reply:
Once we agree on the renaming, given that it would require fixing the`make update` step and rewrite some commit messages, I would do a proper rebase, also squashing the stack of small fixup commits that piled up till now.

But I will have to wait until tomorrow to act upon this. 

I totally favor a renaming, for the reason above, let's just agree on a proper name. 

@bbbrumley/@mattcaswell would you care to chime in with your opinion on proper names?

Reply:
> What about ec_scalarmul_ladder?

Great! Except maybe ec_scalar_mul_ladder? Just pick either :-)

Reply:
> maybe ec_scalar_mul_ladder

+1

Reply:
`ec_scalar_mul_ladder` works for me.